hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,294 @@
+        else: 
+            self._remove_all_vlans_port(port)
+
+        #x = "set interfaces et-0/0/08:0 unit 0 family ethernet-switching interface-mode trunk"
+
+    def _get_native_vlan(self, port):
+        """ Returns the id of the native vlan for the trunked port 
+        if it exists returns nothing otherwise.  
+        """ 
+        port_info = self._interface_info(port)
+        return port_info[port]['native_vlan']
+
+    def _set_native_vlan(self, port, network_id):
+       """Sets native vlan for a trunked port. """
+       set_native_vlan = """
+       set interface {port} native-vlan-id {network_id}
+        """.format(port=port, network_id=network_id)
+       self._set_commit_config(set_native_vlan)
+
+    def _remove_native_vlan(self, port, network_id):
+        """Removes native_vlan tag from port. """
+
+        remove_native_vlan = """
+        delete interface {port} native-vlan-id {network_id}
+        """.format(port=port)
+        self._set_commit_config(remove_native_vlan)
+
+
+                
+    def _add_vlan_to_trunk(self, port, vlan_id):
+        """ Adds vlans to a trunk port. """
+        #eg: set interfaces et-0/0/08:0 unit 0 family ethernet-switching vlan members 200
+
+        add_vlan = """
+        set interfaces {port} unit 0 family ethernet-switching vlan members {vlan_id}
+        """.format(port=port, vlan_id=vlan_id)
+
+        self._set_commit_config(add_vlan)
+
+    def _remove_vlan_from_port(self, port, vlan_id):
+        """ removes a single vlan specified by `vlan_id` """
+        remove_vlan = """
+        delete interfaces {port} unit 0 family ethernet-switching vlan members {vlan_id}
+        """.format(port=port, vlan_id=vlan_id)
+        self._set_commit_config(remove_vlan)
+        
+    def _remove_all_vlans_port(self, port):
+        """ Removes all vlans from the port, including the native vlan.
+        Also converts the interface to access mode. 
+        """
+        remove_all_vlans = """
+            delete interfaces {port} unit 0 family ethernet-switching vlan
+            delete interfaces {port} unit 0 family ethernet-switching interface-mode
+            delete interfaces {port} native-vlan-id
+            """.format(port=port)
+        self._set_commit_config(remove_all_vlans)
+
+    def _get_port_networks(self, port):
+        """ Returns list of all the vlans shared with the port. """
+        port_info = self._interface_info(port)
+        return port_info[port]['vlans'] 
+
+    def _set_port_state(self, port, state):
+        """ Disables or enables the `port`. 
+        Depending on the value of `state`.
+        """
+
+        if state == "enable":
+            set_port_state = """
+            delete interface {port} disable
+            """.format(port=port)
+        elif state == "disable":
+            set_port_state = """
+            set interface {port} disable
+            """.format(port=port)
+        self._set_commit_config(remove_vlan)
```

Comment:
what is `remove_vlan` here? shouldn't we do `self._set_commit_config(set_port_state)`

Reply:
this function is removed from the driver.